typescript/sdk/src/hook/EvmHookReader.ts

```diff
@@ -85,33 +86,46 @@ export class EvmHookReader implements HookReader {
-    const hook = IPostDispatchHook__factory.connect(address, this.provider);
-    const onchainHookType: OnchainHookType = await hook.hookType();
-    this.logger.debug('Deriving HookConfig', { address, onchainHookType });
-
-    switch (onchainHookType) {
-      case OnchainHookType.ROUTING:
-        return this.deriveDomainRoutingConfig(address);
-      case OnchainHookType.AGGREGATION:
-        return this.deriveAggregationConfig(address);
-      case OnchainHookType.MERKLE_TREE:
-        return this.deriveMerkleTreeConfig(address);
-      case OnchainHookType.INTERCHAIN_GAS_PAYMASTER:
-        return this.deriveIgpConfig(address);
-      case OnchainHookType.FALLBACK_ROUTING:
-        return this.deriveFallbackRoutingConfig(address);
-      case OnchainHookType.PAUSABLE:
-        return this.derivePausableConfig(address);
-      case OnchainHookType.PROTOCOL_FEE:
-        return this.deriveProtocolFeeConfig(address);
-      // ID_AUTH_ISM could be OPStackHook, ERC5164Hook or LayerZeroV2Hook
-      // For now assume it's OP_STACK
-      case OnchainHookType.ID_AUTH_ISM:
-        return this.deriveOpStackConfig(address);
-      default:
-        throw new Error(
-          `Unsupported HookType: ${OnchainHookType[onchainHookType]}`,
-        );
+    try {
+      const hook = IPostDispatchHook__factory.connect(address, this.provider);
+      this.logger.debug('Deriving HookConfig', { address });
+
+      // Temporarily turn off SmartProvider logging
+      // Provider errors are expected because deriving will call methods that may not exist in the Bytecode
+      this.setSmartProviderLogLevel('silent');
+      const onchainHookType = await hook.hookType();
+
+      switch (onchainHookType) {
+        case OnchainHookType.ROUTING:
+          return this.deriveDomainRoutingConfig(address);
+        case OnchainHookType.AGGREGATION:
+          return this.deriveAggregationConfig(address);
+        case OnchainHookType.MERKLE_TREE:
+          return this.deriveMerkleTreeConfig(address);
+        case OnchainHookType.INTERCHAIN_GAS_PAYMASTER:
+          return this.deriveIgpConfig(address);
+        case OnchainHookType.FALLBACK_ROUTING:
+          return this.deriveFallbackRoutingConfig(address);
+        case OnchainHookType.PAUSABLE:
+          return this.derivePausableConfig(address);
+        case OnchainHookType.PROTOCOL_FEE:
+          return this.deriveProtocolFeeConfig(address);
+        // ID_AUTH_ISM could be OPStackHook, ERC5164Hook or LayerZeroV2Hook
+        // For now assume it's OP_STACK
+        case OnchainHookType.ID_AUTH_ISM:
+          return this.deriveOpStackConfig(address);
+        default:
+          throw new Error(
+            `Unsupported HookType: ${OnchainHookType[onchainHookType]}`,
+          );
+      }
+    } catch (e) {
+      this.logger.debug(
+        `Hook deriving failed for hook: ${address} with error ${e}`,
+      );
+      return {} as DerivedHookConfig;
```

Comment:
@paulbalaji this changes the existing behavior of derive. Previously, this throws, but now we return an empty HookConfig. can you think of any issues this could cause?

Reply:
I'm onboard with returning an object instead of throwing an error, but `{}` is not a valid `DerivedHookConfig` so we may hit problems where we're trying to read a field we expect e.g. `address` on an empty object

I'd be more in favour of changing the function signature to return `Promise<DerivedHookConfig | undefined>`, and then updating callers of `deriveHookConfig` to handle it how they see fit

(and ditto for the other readers in the follow-up PR)

Reply:
++